src/contrib/coding-style.rst

```diff
@@ -1,12 +1,494 @@
+    #define SSSDBG_CRIT_FAILURE   0x0020   /* level 1 */
+    #define SSSDBG_OP_FAILURE     0x0040   /* level 2 */
+    #define SSSDBG_MINOR_FAILURE  0x0080   /* level 3 */
+    #define SSSDBG_CONF_SETTINGS  0x0100   /* level 4 */
+    #define SSSDBG_FUNC_DATA      0x0200   /* level 5 */
+    #define SSSDBG_TRACE_FUNC     0x0400   /* level 6 */
+    #define SSSDBG_TRACE_LIBS     0x1000   /* level 7 */
+    #define SSSDBG_TRACE_INTERNAL 0x2000   /* level 8 */
+    #define SSSDBG_TRACE_ALL      0x4000   /* level 9 */
+    #define SSSDBG_BE_FO          0x8000   /* level 9 */
+    #define SSSDBG_TRACE_LDB     0x10000   /* level 10 */
+    #define SSSDBG_IMPORTANT_INFO SSSDBG_OP_FAILURE
+
+Macros
+------
+If a macro is safe (for example a simple wrapping function), then the case
+can be lower-case. Macros that are unsafe should be in upper-case.
+This also applies to macros that span multiple lines:
+
+.. code-block:: c
+
+    #define MY_MACRO(a, b) do {   \
+    foo((a) + (b));           \
+    bar(a);                   \
+    } while (0)
+
+Notice that arguments should be in parentheses if there’s a risk. Also notice
+that a is referenced two times, and hence the macro is dangerous. Wrapping the
+body in do { } while (0) makes it safe to use it like this:
+
+.. code-block:: c
+
+    if (expr)
+        MY_MACRO(x, y);
+
+Notice the semicolon is used after the invocation, not in the macro definition.
+
+goto
+====
+We use goto to simplify cleanup operations and some other tasks that need
+to be done before leaving the function. Never use goto to jump backwards
+in the code. Do not use more than one goto label per function.
+
+Common goto labels we are using are:
+
+- done:
+    Label done is used as jump target before exit. Clean-up operations, such as
+    freeing local talloc context, usually follow the done label. Both successful
+    and unsuccessful function executions pass this label.
+
+- fail:
+    Used as special exit path when function fails. Successful function execution
+    typically does not execute statements after this label.
+
+- immediate:
+    The immediate label is used in tevent’s _send functions.
+    The typical usage would look like this:
+
+.. code-block:: c
+
+    if (ret != EOK) {
+        DEBUG(...);
+        goto immediate;
+    }
+
+    immediate:
+    if (ret == EOK) {
+        tevent_req_done(req);
+    } else {
+        tevent_req_error(req, ret);
+    }
+
+    tevent_req_post(req, ev);
+    return req;
+
+Variables
+=========
+- Never use Hungarian notation when naming variables.
+- Use low case multi word underscore separated notation for naming variables.
+- Make variable name meaningful.
+- Always declare variables at the top of the function or block. If you find
+  yourself declaring many variables inside inner block or loop, consider
+  refactoring the block into helper function.
+- One declaration per line is preferred.
+- Initialize loval variables at declaration time when possible.
```

Comment:
s/loval/lval/ ?

Reply:
It mean to be "local variables" :) Fixed

Reply:
Fixed